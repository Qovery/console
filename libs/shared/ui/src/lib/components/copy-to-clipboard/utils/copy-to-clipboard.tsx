export function copyToClipboard(str: string): void {
  navigator.clipboard.writeText(str)
}
