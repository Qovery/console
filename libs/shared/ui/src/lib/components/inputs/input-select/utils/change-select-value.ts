import { act, fireEvent, getByRole, getByTestId } from '@testing-library/react'

export function simulateSpaceKeyClick(element: Element) {
  fireEvent.keyDown(element, { key: ' ' })
  fireEvent.keyUp(element, { key: ' ' })
}

export async function changeSelectValueByLabel(
  baseElement: HTMLElement,
  selectTestId: string,
  value: string
): Promise<void> {
  simulateSpaceKeyClick(getByTestId(getByTestId(baseElement, selectTestId), 'input-select-button'))

  const itemToSelect = getByRole(baseElement, 'option', {
    name: value,
  })
  await act(() => {
    fireEvent.click(itemToSelect)
  })
}
