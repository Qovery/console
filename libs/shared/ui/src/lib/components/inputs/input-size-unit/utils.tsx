export const inputSizeUnitRules = (maxSize?: number, minSize = 0) => ({
  required: 'Please enter a size.',
  validate: (value: number) => (maxSize ? value <= maxSize : undefined),
  max: maxSize,
  min: minSize,
  pattern: {
    value: /^[0-9]+$/,
    message: 'Please enter a number.',
  },
})
