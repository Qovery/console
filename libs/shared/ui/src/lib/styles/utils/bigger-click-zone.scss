.bigger-click-zone {
  position: relative;

  &:after {
    content: '';
    position: absolute;
    inset: -4px;
    // background: rgba(red, .3);
    display: block;
  }
}
