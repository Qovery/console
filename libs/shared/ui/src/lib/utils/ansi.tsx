// eslint-disable-next-line @typescript-eslint/no-restricted-imports
import AnsiPrimitive from 'ansi-to-react'
import { type ComponentProps } from 'react'
import { twMerge } from '@qovery/shared/util-js'

interface AnsiProps extends ComponentProps<typeof AnsiPrimitive> {}

export function Ansi({ className, children, ...props }: AnsiProps) {
  return (
    <AnsiPrimitive className={twMerge('code-ansi', className)} linkify {...props}>
      {children}
    </AnsiPrimitive>
  )
}
