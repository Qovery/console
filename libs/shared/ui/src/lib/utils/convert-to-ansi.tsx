import Ansi from 'ansi-to-react'
import { ReactNode } from 'react'

export const convertToAnsi = (content = ''): ReactNode => {
  return (
    <Ansi className="code-ansi" linkify>
      {content}
    </Ansi>
  )
}
