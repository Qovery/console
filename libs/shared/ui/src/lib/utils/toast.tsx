import { Toast, toast as toastAction } from 'react-hot-toast'
import { ToastContent } from '../components/toast/toast'

export enum ToastEnum {
  SUCCESS = 'SUCCESS',
  ERROR = 'ERROR',
  WARNING = 'WARNING',
}

export const toast = (
  status: ToastEnum,
  title: string,
  description?: string,
  callback?: () => void,
  iconAction?: string,
  labelAction?: string,
  externalLink?: string
) => {
  return toastAction.success((options: Toast) =>
    ToastContent(status, options, title, description, callback, iconAction, labelAction, externalLink)
  )
}

export default toast
