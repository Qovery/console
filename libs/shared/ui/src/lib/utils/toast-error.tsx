import { type SerializedError } from '@qovery/shared/utils'
import toast, { ToastEnum } from './toast'

export function toastError(error: SerializedError | Error, title?: string, message?: string): void {
  toast(ToastEnum.ERROR, title || error.name || 'Error', message || error.message || 'No message found')
}
