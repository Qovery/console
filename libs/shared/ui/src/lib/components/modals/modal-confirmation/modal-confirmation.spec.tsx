import { render } from '__tests__/utils/setup-jest'
import { fireEvent, screen } from '@testing-library/react'

import ModalConfirmation, { ModalConfirmationProps } from './modal-confirmation'

describe('ModalConfirmation', () => {
  let props: ModalConfirmationProps

  beforeEach(() => {
    props = {
      title: 'my title',
      description: 'my description',
      name: 'staging',
      callback: jest.fn(),
    }
  })

  it('should render successfully', () => {
    const { baseElement } = render(<ModalConfirmation {...props} />)
    expect(baseElement).toBeTruthy()
  })

  it('should have an action to copy the name', () => {
    props.name = 'production'

    render(<ModalConfirmation {...props} />)

    Object.assign(window.navigator, {
      clipboard: {
        writeText: jest.fn().mockImplementation(() => Promise.resolve()),
      },
    })

    const copy = screen.queryByTestId('copy-cta') as HTMLSpanElement
    fireEvent.click(copy)

    expect(window.navigator.clipboard.writeText).toHaveBeenCalledWith('production')
  })
})
