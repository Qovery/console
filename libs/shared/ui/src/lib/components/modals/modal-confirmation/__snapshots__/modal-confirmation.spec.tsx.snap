// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalConfirmation should match confirm mode with description 1`] = `
<div>
  <div
    class="p-6"
  >
    <h2
      class="h4 mb-2 max-w-sm text-neutral-400 dark:text-neutral-50"
    >
      my title
    </h2>
    <div
      class="mb-6 text-sm text-neutral-350 dark:text-neutral-50"
    >
      my description
      <span
        class="link relative -top-0.5 ml-1 inline max-w-[250px] cursor-pointer truncate text-sm text-sky-500"
        data-state="closed"
        data-testid="copy-cta"
      >
        staging
         
        <i
          aria-hidden="true"
          class="fa-solid fa-copy "
        />
      </span>
    </div>
    <form>
      <div
        class="mb-5"
        data-testid="input-small-wrapper"
      >
        <div
          class="input input--small flex-grow"
          data-testid="input"
        >
          <label
            class="hidden"
          />
          <input
            class="absolute left-0 top-0 h-full w-full rounded px-2 text-sm text-neutral-400 placeholder:text-neutral-350"
            data-testid="input-value"
            name="name"
            placeholder="Enter the current name"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="flex justify-end gap-3"
      >
        <button
          class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 hover:[&:not(:focus-visible):not(:disabled)]:shadow-sm active:[&:not(:focus-visible)]:shadow-none outline-brand-500 text-sm h-11 px-5 rounded border border-transparent text-neutral-350 hover:[&:not(:active):not(:focus-visible)]:border-neutral-300 hover:[&:not(:disabled)]:text-neutral-400 dark:hover:[&:not(:disabled)]:text-neutral-250 focus-visible:text-neutral-400 active:bg-neutral-150 dark:active:bg-transparent active:text-neutral-400 disabled:text-neutral-300 disabled:bg-neutral-150 disabled:border-none"
          type="button"
        >
          Cancel
        </button>
        <button
          class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-neutral-500 text-sm h-11 px-5 rounded bg-brand-500 active:bg-brand-600 hover:bg-brand-400 text-white disabled:text-brand-300 disabled:bg-brand-100"
          type="submit"
        >
          Confirm
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`ModalConfirmation should match delete mode with description 1`] = `
<div>
  <div
    class="p-6"
  >
    <h2
      class="h4 mb-2 max-w-sm text-neutral-400 dark:text-neutral-50"
    >
      my title
    </h2>
    <div
      class="mb-6 text-sm text-neutral-350 dark:text-neutral-50"
    >
      my description
    </div>
    <form>
      <div
        class="mb-5"
        data-testid="input-small-wrapper"
      >
        <div
          class="input input--small flex-grow"
          data-testid="input"
        >
          <label
            class="hidden"
          />
          <input
            class="absolute left-0 top-0 h-full w-full rounded px-2 text-sm text-neutral-400 placeholder:text-neutral-350"
            data-testid="input-value"
            name="name"
            placeholder="Enter "delete""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="flex justify-end gap-3"
      >
        <button
          class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 hover:[&:not(:focus-visible):not(:disabled)]:shadow-sm active:[&:not(:focus-visible)]:shadow-none outline-brand-500 text-sm h-11 px-5 rounded border border-transparent text-neutral-350 hover:[&:not(:active):not(:focus-visible)]:border-neutral-300 hover:[&:not(:disabled)]:text-neutral-400 dark:hover:[&:not(:disabled)]:text-neutral-250 focus-visible:text-neutral-400 active:bg-neutral-150 dark:active:bg-transparent active:text-neutral-400 disabled:text-neutral-300 disabled:bg-neutral-150 disabled:border-none"
          type="button"
        >
          Cancel
        </button>
        <button
          class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-neutral-500 text-sm h-11 px-5 rounded bg-red-500 active:bg-red-600 hover:bg-red-400 text-white disabled:text-red-300 disabled:bg-red-100"
          type="submit"
        >
          Confirm
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`ModalConfirmation should match delete mode without description 1`] = `
<div>
  <div
    class="p-6"
  >
    <h2
      class="h4 mb-2 max-w-sm text-neutral-400 dark:text-neutral-50"
    >
      my title
    </h2>
    <div
      class="mb-6 text-sm text-neutral-350 dark:text-neutral-50"
    >
      To confirm the deletion of 
      <strong>
        staging
      </strong>
      , please type "delete"
    </div>
    <form>
      <div
        class="mb-5"
        data-testid="input-small-wrapper"
      >
        <div
          class="input input--small flex-grow"
          data-testid="input"
        >
          <label
            class="hidden"
          />
          <input
            class="absolute left-0 top-0 h-full w-full rounded px-2 text-sm text-neutral-400 placeholder:text-neutral-350"
            data-testid="input-value"
            name="name"
            placeholder="Enter "delete""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="flex justify-end gap-3"
      >
        <button
          class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 hover:[&:not(:focus-visible):not(:disabled)]:shadow-sm active:[&:not(:focus-visible)]:shadow-none outline-brand-500 text-sm h-11 px-5 rounded border border-transparent text-neutral-350 hover:[&:not(:active):not(:focus-visible)]:border-neutral-300 hover:[&:not(:disabled)]:text-neutral-400 dark:hover:[&:not(:disabled)]:text-neutral-250 focus-visible:text-neutral-400 active:bg-neutral-150 dark:active:bg-transparent active:text-neutral-400 disabled:text-neutral-300 disabled:bg-neutral-150 disabled:border-none"
          type="button"
        >
          Cancel
        </button>
        <button
          class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-neutral-500 text-sm h-11 px-5 rounded bg-red-500 active:bg-red-600 hover:bg-red-400 text-white disabled:text-red-300 disabled:bg-red-100"
          type="submit"
        >
          Confirm
        </button>
      </div>
    </form>
  </div>
</div>
`;
